test/e2e/storage/testsuites/base.go

```diff
@@ -192,9 +192,22 @@ func (r *genericVolumeTestResource) cleanupResource(driver drivers.TestDriver, p
 	volType := pattern.VolType
 
 	if r.pvc != nil || r.pv != nil {
-		By("Deleting pv and pvc")
-		if errs := framework.PVPVCCleanup(f.ClientSet, f.Namespace.Name, r.pv, r.pvc); len(errs) != 0 {
-			framework.Failf("Failed to delete PVC or PV: %v", utilerrors.NewAggregate(errs))
+		switch volType {
+		case testpatterns.PreprovisionedPV:
+			By("Deleting pv and pvc")
+			if errs := framework.PVPVCCleanup(f.ClientSet, f.Namespace.Name, r.pv, r.pvc); len(errs) != 0 {
+				framework.Failf("Failed to delete PVC or PV: %v", utilerrors.NewAggregate(errs))
+			}
+		case testpatterns.DynamicPV:
+			By("Deleting pvc")
+			// We only delete the PVC so that PV (and disk) can be cleaned up by dynamic provisioner
+			if r.pv.Spec.PersistentVolumeReclaimPolicy != v1.PersistentVolumeReclaimDelete {
+				framework.Failf("Persistent Volume %v specified with reclaim policy that isnt %v", r.pv.Name, v1.PersistentVolumeReclaimDelete)
```

Comment:
Maybe the error message should say it's not supported with the framework currently.

Reply:
done